pkg/util/mount/nsenter_mount.go

```diff
@@ -299,18 +305,30 @@ func (mounter *NsenterMounter) PrepareSafeSubpath(subPath Subpath) (newHostPath
 
 func (mounter *NsenterMounter) SafeMakeDir(subdir string, base string, perm os.FileMode) error {
 	fullSubdirPath := filepath.Join(base, subdir)
+	evaluatedBase, err := mounter.evalHostSymlinks(base, true /* mustExist */)
+	if err != nil {
+		return fmt.Errorf("error resolving symlinks in %s: %s", base, err)
+	}
+	evaluatedBase = filepath.Clean(evaluatedBase)
 	evaluatedSubdirPath, err := mounter.evalHostSymlinks(fullSubdirPath, false /* mustExist */)
 	if err != nil {
 		return fmt.Errorf("error resolving symlinks in %s: %s", fullSubdirPath, err)
 	}
-	kubeletSubdirPath := mounter.getKubeletPath(evaluatedSubdirPath)
-
-	evaluatedBase, err := mounter.evalHostSymlinks(base, true /* mustExist */)
-	if err != nil {
-		return fmt.Errorf("error resolving symlinks in %s: %s", base, err)
+	evaluatedSubdirPath = filepath.Clean(evaluatedSubdirPath)
+
+	rootDir := filepath.Clean(mounter.rootDir)
+	if pathWithinBase(evaluatedBase, rootDir) {
+		// Base is in /var/lib/kubelet. This directory is shared between the
+		// container with kubelet and the host. We don't need to add '/rootfs'.
+		// This is useful when /rootfs is mounted as read-only - we can still
+		// create subpaths for paths in /var/lib/kubelet.
+		doSafeMakeDir(evaluatedSubdirPath, evaluatedBase, perm)
 	}
-	kubeletBase := mounter.getKubeletPath(evaluatedBase)
 
+	// Base is somewhere on the host's filesystem. Add /rootfs and try to make
+	// the directory there.
```

Comment:
Add comment that this requires /rootfs to be writeable.

Reply:
added